geo/src/algorithm/coordinate_position.rs

```diff
@@ -0,0 +1,749 @@
+use crate::algorithm::{
+    bounding_rect::BoundingRect, dimensions::HasDimensions, intersects::Intersects,
+    kernels::HasKernel,
+};
+use crate::{
+    Coordinate, Geometry, GeometryCollection, Line, LineString, MultiLineString, MultiPoint,
+    MultiPolygon, Point, Polygon, Rect, Triangle,
+};
+
+/// The position of a `Coordinate` relative to a `Geometry`
+#[derive(PartialEq, Clone, Debug)]
+pub enum CoordPos {
```

Comment:
moved this from utils - reusing form the "coord_pos_relative_to_ring".

@rmanoka - do you see any issue with re-using this enum from coord_pos_relative_to_ring? 

At first I introduced a new "Interior/Exterior/Boundary" enum, but it seemed redundant.

Reply:
Sounds fine to me